app/code/Magento/CatalogGraphQl/Model/Resolver/Products/DataProvider/Related/AbstractDataProvider.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\Resolver\Products\DataProvider\Related;
+
+abstract class AbstractDataProvider
```

Comment:
Pls, don't use supertypes only for code reusing (if a relation is not like `type A is subtype of B`)